ts/sagas/wallet.ts

```diff
@@ -24,12 +77,161 @@ function* fetchTransactions(
 }
 
 function* fetchCreditCards(
-  loadCards: () => Promise<ReadonlyArray<CreditCard>>
+  loadCards: () => Promise<ReadonlyArray<Wallet>>
 ): Iterator<Effect> {
-  const cards: ReadonlyArray<CreditCard> = yield call(loadCards);
+  const cards: ReadonlyArray<Wallet> = yield call(loadCards);
   yield put(cardsFetched(cards));
 }
 
+const navigateTo = (routeName: string, params?: object) => {
+  return NavigationActions.navigate({ routeName, params });
+};
+
+function* paymentSaga(): Iterator<Effect> {
+  yield put(navigateTo(ROUTES.PAYMENT_SCAN_QR_CODE)); // start by showing qr code scanner
+
+  while (true) {
+    const action = yield take([
+      PAYMENT_INSERT_DATA_MANUALLY,
+      PAYMENT_SHOW_SUMMARY,
+      PAYMENT_CONFIRM_SUMMARY,
+      PAYMENT_PICK_PAYMENT_METHOD,
+      PAYMENT_CONFIRM_PAYMENT_METHOD,
+      PAYMENT_REQUEST_OTP,
+      PAYMENT_VERIFY_OTP,
```

Comment:
what if the verify fails and I want another OTP?

Reply:
The behavior triggered when you click on the "resend otp" link hasn't been implemented yet, but in short doing so will dispatch a "requestOtp" action (which is also dispatched upon first displaying the "enter OTP screen"). This action requests a new OTP to pagoPA, and when the user enters it and clicks "submit", the OTP sent will be the last one entered by the user (this -- of course -- is how I see things unfolding, but the OTP request/check process is yet to be analyzed with the pagoPA team)